### ISSUE TITLE
How to access a given service url within operators code

### ISSUE THREAD

**Issue by sjmittal** [ContainerSolutions/java-operator-sdk#109]
Hi,
I have a service described as:

```
Name:              objectstore-service
Namespace:         objectstore-runtime
Labels:            <none>
Annotations:       <none>
Selector:          app=objectstore
Type:              ClusterIP
IP:                10.106.97.40
Port:              http-minio  9000/TCP
TargetPort:        9000/TCP
Endpoints:         172.18.0.7:9000
Session Affinity:  None
Events:            <none>
```

However when I try to access is service url as per example:
https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/ServiceExample.java

using something like:
`kubernetesClient.services().inNamespace("objectstore-runtime").withName("objectstore-service").getURL("http-minio")`

I get null.

Is there some other way to access the service.

My usecase is basically to access the service url via HTTP and invoke some REST endpoints to do some updates.

**Comment by csviri** [comment]
Hi @sjmittal, this should be the right way, but this part should be handled fully by kubernetes fabric8 client. If you are in the right namespace and can access the service this should work. If not please address the question in the fabric8 java client project:
https://github.com/fabric8io/kubernetes-client

**Comment by adam-sandor** [comment]
I don't really understand what that getURL method is trying to do. Services are basically TCP-level loadbalancers, so they don't really deal with HTTP URLs. You can get the service object by `kubernetesClient.services().inNamespace("objectstore-runtime").withName("objectstore-service").get()` and grab the IP and port from there. Adding the HTTP path will have to be done in your logic based on what your HTTP service exposes.

**Comment by sjmittal** [comment]
I have logged an issue here:
https://github.com/fabric8io/kubernetes-client/issues/2233
How I understand is that not all services are of type `LoadBalancer`.

I think `fabric8` client api provides a helper method to get the URL of service via port name.
However this helper method does not consider service of type `ClusterIP`.

In my operator I need to connect to my running container and I have defined a service of type `ClusterIP`. 
I was thinking that `getURL("port-name")` would help me get that URL to connect to.

However as mentioned above grabbing IP and port directly from the created instance of service is sufficient and there is no need to rely on helper method provided by the service.

This can be closed as its not related to java-operator-sdk.

.
